src/tritonserver.cc

```diff
@@ -2573,13 +2573,12 @@ TRITONSERVER_ServerStop(TRITONSERVER_Server* server)
 }
 
 TRITONAPI_DECLSPEC TRITONSERVER_Error*
-TRITONSERVER_ServerStopWithTimeout(
+TRITONSERVER_ServerSetExitTimeout(
```

Comment:
Why do we need both `TRITONSERVER_ServerOptionsSetExitTimeout` and `TRITONSERVER_ServerSetExitTimeout`?

Reply:
`TRITONSERVER_ServerOptionsSetExitTimeout` is setting the timeout on an opaque object which is used when creating a new server object.
`TRITONSERVER_ServerSetExitTimeout` is to set the exit timeout on an exiting server object.

Reply:
So we're using this in server PR to logically separate the two parts of shutdown, but overall respect the --exit-timeout setting originally set?

ex: 30 sec timeout

- wait 30 secs for connections
- update core timeout to 0 seconds
- core shutdown exit immediately, no waiting

or

- wait 20 seconds connections
- update core timeout to 10 seconds for sum total of 30
- core shutdown

?

Reply:
> overall respect the --exit-timeout setting originally set?

Yes, we respect the `--exit-timeout` command line option value overall.

The value is passed to the core with `TRITONSERVER_ServerOptionsSetExitTimeout` opaque option object when the core starts, but it is no longer accurate by the time of shutdown, because part of it can be consumed by the server frontend shutdown which happens ahead of the core shutdown, so it is always updated immediately with `TRITONSERVER_ServerSetExitTimeout` before the core shutdown.